src/sqlfluff/core/rules/analysis/select_crawler.py

```diff
@@ -120,6 +120,8 @@ def get_wildcard_info(self) -> List[WildcardInfo]:
                             seg,
                             [
                                 alias_info.ref_str
+                                if alias_info.aliased
```

Comment:
This fixes the real issue: The rule was confusing two things:
* A table whose base name was "blah"
* A CTE named "blah"